submissions/salsita/src/reducers/sithLoadingReducer.js

```diff
@@ -0,0 +1,75 @@
+import { Range } from 'immutable';
+
+import buildMessage from '../messageBuilder';
+import * as Effects from '../constants/effects';
+import * as SithStates from '../constants/sithStates';
+import * as Configuration from '../constants/configuration';
+import { SithRecord } from '../constants/records';
+
+// This simply initializes the application state and triggers loading of first Sith
+export const applicationMounting = reduction => reduction.withMutations(mutableReduction => {
+  mutableReduction
+    .setIn(['appState', 'disabledUp', true])
+    .setIn(['appState', 'disabledDown', true])
+    .setIn(['appState', 'scrollingDown'], true)
+    .setIn(['appState', 'sithsSpotted'], false)
+    .setIn(['appState', 'siths'], new Range(0, Configuration.SITH_SLOTS).map(() => new SithRecord()).toList())
+    .setIn(['appState', 'siths', 0, 'id'], Configuration.INITIAL_SITH_TO_LOAD)
+    .setIn(['appState', 'siths', 0, 'state'], SithStates.LOADING)
+    .setIn(['appState', 'planet'], null)
+    .update('effects', effects => effects
+      .push(buildMessage(Effects.API_LOAD_SITH, {
+        sithId: Configuration.INITIAL_SITH_TO_LOAD,
+        scrollingDown: true
+      }))
+      .push(buildMessage(Effects.API_CONNECT_WS)));
+});
+
+export const sithLoaded = (reduction, payload) => {
+  const loadedSithIndex = reduction.getIn(['appState', 'siths']).findIndex(sith => sith.get('id') === payload.sith.id);
+  const scrollingDown = payload.scrollingDown;
+  const apprentice = payload.sith.apprentice.id;
+  const master = payload.sith.master.id;
+  const sithName = payload.sith.name;
```

Comment:
I'd probably say:

```
const { apprentice, master, name: sithName } = payload.sith;
```

Obviously in the subsequent code you'll use `apprentice.id` and `master.id`.